src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -543,6 +553,10 @@ static int do_map(int argc, const char *argv[])
   common_init_finish(g_ceph_context);
   global_init_chdir(g_ceph_context);
 
+  init_async_signal_handler();
+  register_async_signal_handler_oneshot(SIGINT, handle_signal);
+  register_async_signal_handler_oneshot(SIGTERM, handle_signal);
```

Comment:
I think you register to early (and unregister too late), at this moment e.g. devpath still might be unknown (and nbd not opened), so if a signal happens before we start listening on nbd, disconnect request is lost.

I suggest registering the handlers just before `ioctl(nbd, NBD_DO_IT)` and unregister just after.